File: 프로그래머스/2/42586. 기능개발/기능개발.py
def solution(progresses, speeds):
    answer = []
    
    idx=0
    while True:
        for i in range(len(progresses)):
            progresses[i]+=speeds[i]
        cnt=0
        for i in range(idx,len(progresses)):
            if progresses[i]>=100:
                cnt+=1
            else:
                break
        if cnt>0:
            answer.append(cnt)
            idx+=cnt
        if sum(answer)==len(progresses):
            return answer
    
            

File: 프로그래머스/2/42586. 기능개발/기능개발.java
import java.util.*;

class Solution {
    public List<Integer> solution(int[] progresses, int[] speeds) {
        int[] answer = {};
        Deque<Integer> arr = new ArrayDeque<>();
        for(int i = 0;i<progresses.length;i++){
            double temp = Math.ceil((double)(100-progresses[i])/speeds[i]);
            arr.addLast((int)temp);
        }
        List<Integer> list = new ArrayList<>();
        while(!arr.isEmpty()){
            int cnt=1;
            int popItem=arr.pop();
            while(!arr.isEmpty() && popItem >= arr.peek()){
                arr.remove();
                cnt++;
            }
            list.add(cnt);
        }
        return list;
    }
}